### PR TITLE
add dn_detr algo

### PR DESCRIPTION
**Pull request by tuofeilunhifi**
1. support dn-detr-r50: 44.39
2. refactor set_criterion

### CODE REVIEW

File: configs/detection/_base_/dataset/autoaug_coco_detection.py [hunk 1]
```diff
@@ -119,3 +119,15 @@
 
 data = dict(
     imgs_per_gpu=2, workers_per_gpu=2, train=train_dataset, val=val_dataset)
+
+# evaluation
+eval_config = dict(initial=True, interval=1, gpu_collect=False)
```
Comment (Cathy0908) [inline configs/detection/_base_/dataset/autoaug_coco_detection.py:124]: set initial to False

File: configs/detection/detr/detr.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
 # model settings
 model = dict(
     type='Detection',
-    pretrained=
-    'http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/pretrained_models/easycv/resnet/torchvision/resnet50.pth',
+    pretrained=True,
```
Comment (Cathy0908) [inline configs/detection/detr/detr.py:4]: if user set pretrained=True, and the model does not have a default url, an error should be raised directly

File: docs/source/model_zoo_det.md [hunk 3]
```diff
@@ -19,14 +19,10 @@ Pretrained on COCO2017 dataset.
...
-| DAB-DETR-r50    | [dab-detr-r50](https://github.com/alibaba/EasyCV/tree/master/configs/detection/dab_detr/dab_detr_r50_8x2_50e_coco.py) | 42.52                   | 63.03          | [model](https://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/detection/dab_detr/epoch_50.pth) - [log](https://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/detection/dab_detr/20220610_122811.log.json) |
+| DAB-DETR-r50    | [dab-detr-r50](https://github.com/alibaba/EasyCV/tree/master/configs/detection/dab_detr/dab_detr_r50_8x2_50e_coco.py) | 42.52                   | 63.03          | [model](https://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/detection/dab_detr/dab_detr_epoch_50.pth) - [log](https://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/detection/dab_detr/20220610_122811.log.json) |
```
Comment (Cathy0908) [inline docs/source/model_zoo_det.md:27]: add Params, flops, inference time for added model

File: configs/detection/_base_/dataset/autoaug_coco_detection.py (hunk 1 above)
Comment (tuofeilunhifi) [reply configs/detection/_base_/dataset/autoaug_coco_detection.py]: done

File: configs/detection/detr/detr.py (hunk 2 above)
Comment (tuofeilunhifi) [reply configs/detection/detr/detr.py:4]: done

File: docs/source/model_zoo_det.md (hunk 3 above)
Comment (tuofeilunhifi) [reply docs/source/model_zoo_det.md]: done

**Review by Cathy0908 — APPROVED** [review]

**Pull request merged**